core/src/main/java/org/elasticsearch/common/io/stream/Streamable.java

```diff
@@ -22,11 +22,26 @@
 import java.io.IOException;
 
 /**
+ * Implementers can be written to a {@linkplain StreamOutput} and read from a {@linkplain StreamInput}. This allows them to be "thrown
+ * across the wire" using Elasticsearch's internal protocol. If the implementer also implements equals and hashCode then a copy made by
+ * serializing and deserializing must be equal and have the same hashCode. It isn't required that such a copy be entirely unchanged. For
```

Comment:
Some of these are coming in as part of the query refactoring. It's worth opening an issue for. @colings86 will have opinions because he's been one of the folks writing these tests. If we wrote and enforcer we'd have to be pretty careful to make sure that it used all of the query refactoring's work. We have the skeleton for such an enforcer sitting in the NamingConventionsTest, soon to be moved into a gradle task by #16811.